src/ast/mod.rs

```diff
@@ -435,6 +435,7 @@ pub enum Statement {
     Query(Box<Query>),
     /// INSERT
     Insert {
+        or: Option<crate::dialect::keywords::Keyword>,
```

Comment:
I would encode this as a new struct (named OrAction) which explicitly lists the actions allowed.

Reply:
Thanks for bringing this up, Dan. Judging from https://sqlite.org/lang_conflict.html the more appropriate name would be something like `SqliteConflictResolutionAlgorithm`, as these values can appear in many contexts, not just after `INSERT OR`.

Reply:
Oh and this needs a comment indicating this implements an sqlite-specific`INSERT OR action` to specify a constraint conflict resolution method.

Reply:
`SqliteConflictResolutionAlgorithm` is really a long name, how about `OnConflictAction`  or `OnconflictAlgorithm`?

Reply:
The "Sqlite" bit is pretty important part of the name, as `ON CONFLICT` exists in other dialects, and has very little in common with sqlite's. SqliteOnConflict is fine with me if you really want to make it shorter.